### ISSUE TITLE
Conversion of bmp images to png fails during GLB export in Blender 3.3 Beta

### ISSUE THREAD

**Issue by saturday06** [KhronosGroup/glTF-Blender-IO#1703]
**Describe the bug**
Conversion of bmp images to png fails during GLB export in Blender 3.3 Beta. Using Blender 3.2.2 or previous versions is successful.

**To Reproduce**
1. Create a bitmap image (*.bmp) with Windows Paint App.
2. Open Blender 3.3 Beta
3. Add a 'Plane' mesh
4. Add new "Principled BSDF" material to the mesh.
5. Change it's "Emission" to a Image Texture. Attach the bitmap image.
7. Export GLB.
8. See error

or

1. Extract [bug-report-20220807.zip](https://github.com/KhronosGroup/glTF-Blender-IO/files/9276775/bug-report-20220807.zip)
2. Open the extracted .blend file with Blender 3.3 Beta.
3. Export GLB.

```
Read prefs: C:\Users\saturday06\AppData\Roaming\Blender Foundation\Blender\3.3\config\userpref.blend
Read blend: C:\Users\saturday06\Desktop\bug-report-20220807\bug-report-20220807\bug-report-20220807.blend
11:54:33 | INFO: Draco mesh compression is available, use library at C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\python\lib\site-packages\extern_draco.dll
11:54:37 | INFO: Starting glTF 2.0 export
11:54:37 | INFO: Extracting primitive: Plane
11:54:37 | INFO: Primitives created: 1
Traceback (most recent call last):
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\__init__.py", line 648, in execute
    return gltf2_blender_export.save(context, export_settings)
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_export.py", line 35, in save
    json, buffer = __export(export_settings)
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_export.py", line 52, in __export
    __gather_gltf(exporter, export_settings)
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_export.py", line 66, in __gather_gltf
    active_scene_idx, scenes, animations = gltf2_blender_gather.gather_gltf2(export_settings)
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather.py", line 30, in gather_gltf2
    scenes.append(__gather_scene(blender_scene, export_settings))
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_cache.py", line 38, in wrapper_cached
    result = func(*args, **kwargs)
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather.py", line 69, in __gather_scene
    node = gltf2_blender_gather_nodes.gather_node(
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_nodes.py", line 35, in gather_node
    mesh=__gather_mesh(vnode, blender_object, export_settings),
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_nodes.py", line 246, in __gather_mesh
    result = gltf2_blender_gather_mesh.gather_mesh(blender_mesh,
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_cache.py", line 38, in wrapper_cached
    result = func(*args, **kwargs)
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_mesh.py", line 61, in gather_mesh
    primitives=__gather_primitives(blender_mesh, uuid_for_skined_data, vertex_groups, modifiers, materials, export_settings),
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_mesh.py", line 147, in __gather_primitives
    return gltf2_blender_gather_primitives.gather_primitives(blender_mesh,
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_cache.py", line 38, in wrapper_cached
    result = func(*args, **kwargs)
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_primitives.py", line 83, in gather_primitives
    material = gltf2_blender_gather_materials.gather_material(
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_cache.py", line 38, in wrapper_cached
    result = func(*args, **kwargs)
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_materials.py", line 56, in gather_material
    emissive_texture, uvmap_actives_emissive_texture = __gather_emissive_texture(blender_material, export_settings)
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_materials.py", line 215, in __gather_emissive_texture
    return export_emission_texture(blender_material, export_settings)
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_materials_emission.py", line 54, in export_emission_texture
    emissive_texture, use_actives_uvmap_emissive, _ = gltf2_blender_gather_texture_info.gather_texture_info(emissive, (emissive,), export_settings)
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_texture_info.py", line 23, in gather_texture_info
    return __gather_texture_info_helper(primary_socket, blender_shader_sockets, 'DEFAULT', filter_type, export_settings)
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_cache.py", line 38, in wrapper_cached
    result = func(*args, **kwargs)
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_texture_info.py", line 44, in __gather_texture_info_helper
    index, factor = __gather_index(blender_shader_sockets, export_settings)
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_texture_info.py", line 141, in __gather_index
    return gltf2_blender_gather_texture.gather_texture(blender_shader_sockets, export_settings)
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_cache.py", line 38, in wrapper_cached
    result = func(*args, **kwargs)
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_texture.py", line 31, in gather_texture
    source, factor = __gather_source(blender_shader_sockets, export_settings)
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_texture.py", line 83, in __gather_source
    return gltf2_blender_gather_image.gather_image(blender_shader_sockets, export_settings)
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_cache.py", line 38, in wrapper_cached
    result = func(*args, **kwargs)
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_image.py", line 46, in gather_image
    buffer_view, factor_buffer_view = __gather_buffer_view(image_data, mime_type, name, export_settings)
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_cache.py", line 38, in wrapper_cached
    result = func(*args, **kwargs)
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_image.py", line 106, in __gather_buffer_view
    data, factor = image_data.encode(mime_type)
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_image.py", line 136, in encode
    return self.__encode_happy(), None
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_image.py", line 146, in __encode_happy
    return self.__encode_from_image(self.blender_image())
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_image.py", line 228, in __encode_from_image
    return _encode_temp_image(tmp_image, self.file_format)
  File "C:\Users\saturday06\Desktop\blender-3.3.0-beta\3.3\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_image.py", line 240, in _encode_temp_image
    with open(tmpfilename, "rb") as f:
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\SATURD~1\\AppData\\Local\\Temp\\tmprau94mjh/img'
```

**Expected behavior**

No error.

**Screenshots**

https://user-images.githubusercontent.com/532872/183290636-eee21707-4818-47ac-ae7e-e619feabb881.mp4

**.blend file/ .gltf**

[bug-report-20220807.zip](https://github.com/KhronosGroup/glTF-Blender-IO/files/9276775/bug-report-20220807.zip)

**Version**
 - OS: Windows 11, macOS Monterey
 - Broken: version: 3.3.0 Beta, branch: master, commit date: 2022-08-05 17:53, hash: `rB74f983738ab4`
 - Worked: 3.2.2, 2.83.20

**Comment by julienduroure** [comment]
Hello,
I was first able to reproduce it, but the second time it works. And now, I am not able to reproduce it 100%, even after restarting Blender or reboot.
Seems to be quite random :-/

**Comment by saturday06** [comment]
I found that the bmp file are automatically overwritten with png during the glb export process. Then the error will disappear next time.

**Comment by saturday06** [comment]
I was able to reproduce the error again by deleting the bmp file whose contents are png and recreating a new bmp file.

**Comment by julienduroure** [comment]
Thanks a lot for this very valuable info. Investigation in progress

**Comment by julienduroure** [comment]
Ok, seems the issue comes from Blender API, not from the addon.
I opened the issue on Blender tracker: https://developer.blender.org/T100281

**Comment by julienduroure** [comment]
Seems upstream bug is now fixed.
Let's wait tomorrow's build to test and close this ticket

**Comment by saturday06** [comment]
I also checked. Thank you!

**Comment by aelainelong** [comment]
@julienduroure What was the resolution of this issue? Is there a specific version of Blender that we need to use to get past this? I am also seeing this error when attempting to export to gltf.

**Comment by julienduroure** [comment]
This should be solved in stable 3.3.0 (and any newer) version
